server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java

```diff
@@ -0,0 +1,167 @@
+import org.infinispan.rest.configuration.RestServerConfiguration;
+import org.infinispan.rest.framework.ContentSource;
+import org.infinispan.rest.framework.RestRequest;
+import org.infinispan.rest.framework.RestResponse;
+import org.infinispan.rest.framework.impl.Invocations;
+import org.infinispan.stats.Stats;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+public class CacheResourceV2 extends CacheResource {
+
+   private static final JsonReader JSON_READER = new JsonReader();
+   private static final JsonWriter JSON_WRITER = new JsonWriter();
+   private static final ParserRegistry PARSER_REGISTRY = new ParserRegistry();
+   private final ObjectMapper mapper;
+
+
+   public CacheResourceV2(RestCacheManager<Object> restCacheManager, RestServerConfiguration serverConfig,
+                          ObjectMapper mapper, ThreadPoolExecutor executor) {
+      super(restCacheManager, serverConfig, executor);
+      this.mapper = mapper;
+   }
+
+   @Override
+   public Invocations getInvocations() {
+      return new Invocations.Builder()
+            // Key related operations
+            .invocation().methods(PUT, POST).path("/v2/caches/{cacheName}/{cacheKey}").handleWith(this::putValueToCache)
+            .invocation().methods(GET, HEAD).path("/v2/caches/{cacheName}/{cacheKey}").handleWith(this::getCacheValue)
+            .invocation().method(DELETE).path("/v2/caches/{cacheName}/{cacheKey}").handleWith(this::deleteCacheValue)
+
+            // Info and statistics
+            .invocation().methods(GET, HEAD).path("/v2/caches/{cacheName}/config").handleWith(this::getCacheConfig)
+            .invocation().methods(GET).path("/v2/caches/{cacheName}/stats").handleWith(this::getCacheStats)
+
+            // Cache lifecycle
+            .invocation().methods(POST).path("/v2/caches/{cacheName}").handleWith(this::createCache)
+            .invocation().method(DELETE).path("/v2/caches/{cacheName}").handleWith(this::removeCache)
+
+            // Operations
+            .invocation().methods(GET).path("/v2/caches/{cacheName}").withAction("clear").handleWith(this::clearEntireCache)
+
+            // Search
+            .invocation().methods(GET, POST).path("/v2/caches/{cacheName}").withAction("search").handleWith(queryAction::search)
+            .create();
+
+   }
+
+   private CompletionStage<RestResponse> removeCache(RestRequest request) {
+      NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+      String cacheName = request.variables().get("cacheName");
+      Cache<?, ?> cache = restCacheManager.getCache(cacheName, request.getSubject());
+      if (cache == null) {
+         responseBuilder.status(HttpResponseStatus.NOT_FOUND);
+         return CompletableFuture.completedFuture(responseBuilder.build());
+      }
+      return CompletableFuture.supplyAsync(() -> {
```

Comment:
Shouldn't we be passing an executor here? Was this missed by will's checkstyle PR?

Reply:
You're right, it was a bad rebase, will fix it